google-cloud-spanner/src/test/java/com/google/cloud/spanner/connection/ConnectionOptionsTest.java

```diff
@@ -1329,4 +1329,17 @@ public void testEnableDirectAccess() {
         "spanner://localhost:15000/projects/default/instances/default/databases/singers-db;usePlainText=true;enableDirectAccess=true");
     assertTrue(builderWithDirectPathParam.build().isEnableDirectAccess());
   }
+
+  @Test
+  public void testUniverseDomain() {
+    ConnectionOptions.Builder builderWithNoUniverseDomainParam = ConnectionOptions.newBuilder();
+    builderWithNoUniverseDomainParam.setUri(
+        "spanner://localhost:15000/instances/default/databases/singers-db;usePlainText=true");
+    assertEquals("googleapis.com", builderWithNoUniverseDomainParam.build().getUniverseDomain());
+
+    ConnectionOptions.Builder builderWithUniverseDomainParam = ConnectionOptions.newBuilder();
+    builderWithUniverseDomainParam.setUri(
+        "spanner://localhost:15000/projects/default/instances/default/databases/singers-db;usePlainText=true;universeDomain=abc.goog");
+    assertEquals("abc.goog", builderWithUniverseDomainParam.build().getUniverseDomain());
```

Comment:
Can we also add a verification that the actual host that the connection would try to use in this case is `localhost.abc.goog:15000`? Or would it be something else? (I guess my question is; what does it mean if you set a universe domain and a host name?)